tests/components/tado/test_tado_adapter.py

```diff
@@ -0,0 +1,423 @@
+"""The tado_adapter tests for the tado platform."""
+
+
+from tests.components.tado.mocks import _mock_tado_climate_zone_from_fixture
+
+
+async def test_ac_issue_32294_heat_mode(hass):
+    """Test smart ac cool mode."""
+    ac_issue_32294_heat_mode = await _mock_tado_climate_zone_from_fixture(
```

Comment:
We shouldn't test the device data directly. We should just test the state of the entity from the state machine and the services that the integration provides via the service registry.

Reply:
@MartinHjelmare there are more things wrong in this PR. I should probably have @bdraco create a new PR based on his existing branch and continue there?

Reply:
That's fine. :+1:

Whoever continues should make sure to add the tests first.

Reply:
@michaelarnauts I've got the zone code PR opened for PyTado here with all of your json fixtures (thanks for that!): https://github.com/wmalgadey/PyTado/pull/29

I'm reworking the original HA PR here https://github.com/bdraco/home-assistant/tree/tado_take_2 to consume this add the tests @MartinHjelmare is recommending